### PR TITLE
Multicloud Check for old instances -  POL 32

### PR DESCRIPTION
**Pull request by LegacyAcres**
### Description

Adds a new script to terminate instances that are running longer than X

### Issues Resolved

POL32

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: compliance/long_running_isntances/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Long Running Instances 
+
+
+### What it does
+
+This policy template checks all instances running, and checks them against a user provided number of days.  If the instance has been running longer than the specified number of days then the policy reports those instances for termination, and seeks approval before termination.  The policy accepts using tags to not include instances.
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/README.md]: change "The policy accepts using tags to not include instances." to "The policy accepts using tags to exclude instances."

File: compliance/long_running_isntances/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+### Input Parameters
+
+This policy has the following input parameters required when launching the policy.
+
+- *Days* - How many days is considered long running.
+- *Tags to Ignore* - A comma delimited list of tags that indicate the policy should ignore instances with those tags
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/README.md]: Need to add email param

File: compliance/long_running_isntances/long_running_instances.pt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  description "Threshold of days to consider an instance long running."
+end
+
+parameter "param_tagstoignore" do
+  type "string"
+  label "Tags to Ignore"
+  description "Comma-seperated list of tags to exclude instances from consideration"
+  allowed_pattern /^([a-zA-Z0-9-_=]+:[a-zA-Z0-9-_=]+,*|)+$/
+end
+
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/long_running_instances.pt]: need to add email param.

File: compliance/long_running_isntances/long_running_instances.pt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                  server_json = { "href": href, "days_elapsed": dayselapsed, "name": name, "bootedtimestamp": bootedtimestamp }
+                  final_result.push(server_json)
+              }
+          }
+      }
+  }
+EOS
+end
+
+define termianteinstancesapproved() do
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/long_running_instances.pt]: It's best practice to use underscore _ to separate words in method name.  termiante_instances_approved

You also need to pass *data* to the definition.  did your tests pass

File: compliance/long_running_isntances/long_running_instances.pt [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  foreach $instance in $filterinstances do
+    @instance = rs_cm.get(href: $instance['href'])
+    @instance.terminate()
+  end
+end
+
+# escalation to terminate instances over the limit of days
+escalation "terminateinstances" do
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/long_running_instances.pt]: best practice: use underscore to separate the words termiate_instances

File: compliance/long_running_isntances/long_running_instances.pt [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    label "Escalation approval"
+    description "Do you wish to terminate the volumes that are long running?"
+    parameter "approval_reason" do
+      type "string"
+      label "Reason for approval"
+      description "Explain why this is approved"
+    end
+  
+end
+run "termianteinstancesapproved"
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/long_running_instances.pt]: use underscores, and you need to include the argument for *data*

File: compliance/long_running_isntances/long_running_instances.pt [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+There are {{ len data }} instance(s) discovered by this policy.
+
+| Name |                  HREF |         Days Elapsed | 
+| ---- | ----------------------|----------------------| 
+{{ range data -}}
+| {{.name}} | {{ .href }} | {{.days_elapsed}} | 
+{{ end -}}
+EOS
+  
+  escalate $terminateinstances
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/long_running_instances.pt]: don't forget to rename the definition name to match your underscore change.

File: compliance/long_running_isntances/long_running_instances.pt [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+  #check if empty
```
Comment (cdwilhelm) [inline compliance/long_running_isntances/long_running_instances.pt]: remove unnecessary comments

**Comment by cdwilhelm** [comment]
@LegacyAcres  any progress on getting this finished up so we can release.  It doesn't need much more work.  See my comments to be resolved.

**Comment by LegacyAcres** [comment]
So this branch needs to be deleted.  After we looked into it we changed the POL32 because of limitations, and I need to re-write from scratch.

**Comment by cdwilhelm** [comment]
this was completed in another PR

**Pull request closed without merging**